File: web/src/types/ContentCreator.ts
import Position from './Position'
import UserPic from './UserPic'

type ContentCreator = {
    id: string
    name: string
    surname: string
    nameLatin: string
    surnameLatin: string
    bornPlace: string
    heightCm: number
    personCategory: Position
    personCategory: Position
    userPic: UserPic
    isDead: boolean
    birthDate: string
    deathDate: string
}

export default ContentCreator

File: server/storage/src/main/java/ru/storage/person/content_creator/ContentCreator.java
package ru.storage.person.content_creator;

import com.fasterxml.jackson.annotation.JsonFormat;
import java.time.LocalDate;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;
import ru.storage.person.PersonCategory;
import ru.storage.person.userpic.UserPic;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document
public class ContentCreator {
  @Id private String id;
  private String name;
  private String surname;
  private String nameLatin;
  private String surnameLatin;
  private String bornPlace;
  private Integer heightCm;
  private Integer age;
  private PersonCategory personCategory;
  @DBRef private UserPic userPic;
  private Boolean isDead;

  @JsonFormat(pattern = "dd.MM.yyyy")
  private LocalDate birthDate;

  @JsonFormat(pattern = "dd.MM.yyyy")
  private LocalDate deathDate;
}
